p2p/client.go

```diff
@@ -597,6 +597,11 @@ func (c *Client) retrieveBlockSyncLoop(ctx context.Context, msgHandler BlockSync
 
 				c.logger.Debug("Blocksync block received ", "height", h)
 				msgHandler(&block)
+				state, err := c.store.LoadState()
+				if err != nil {
+					return
+				}
+				h = max(h, state.NextHeight()-1)
```

Comment:
why not setting it in the `for` loop definition?
also need to validate that `h` progress, otherwise we can be stuck in endless loop (in case the height doesn't progress)

Reply:
i think is fine, because the max func will cause that h is always progressing, it will just skip already applied heights. in case state.NextHeight() is updated in parallel